tests/profile.py

```diff
@@ -0,0 +1,584 @@
+    "false") and convert that argument to an actual bool value (True or False).
+    '''
+    string_lower = string.lower()
+    if string_lower == "true":
+        return True
+    elif string_lower == "false":
+        return False
+    raise argparse.ArgumentTypeError('Argument must be True or False')
+
+
+def run_job_profiler(args):
+
+    # ------------------------------------------------
+    # Fetch Job
+    # ------------------------------------------------
+    logger.info("Fetching Job %s", args.jid)
+    client = api_client.ApiClient()
+    r_body, _ = client.make_request('api/v1/jobs',
+                                    params="filter=jid_eq_%s" % args.jid,
+                                    use_api_key=True
+                                    )
+
+    jobs = json.loads(r_body).get("data")
+    if not jobs:
+        raise Exception("No Job %s found" % args.jid)
+    assert len(jobs) < 2, "Found more than 1 Job with jid %s" % args.jid
+    job = jobs[0]
+    upload_id = job["upload"]
+
+    # ------------------------------------------------
+    # Fetch Job's Upload
+    # ------------------------------------------------
+    logger.info("Fetching Upload %s", upload_id)
+    r_body, _ = client.make_request('api/v1/uploads/%s' % upload_id,
+                                    use_api_key=True
+                                    )
+    upload = json.loads(r_body).get("data")
+    filepaths = upload["upload_files"].keys()
+
+    profile(
+        filepaths,
+        stat=args.stat,
+        read=args.read,
+        hash_md5=args.md5,
+        hash_xx=args.xxhash if XXHASH else None,
+        warmup=args.warmup,
+        read_size=args.read_size,
+        thread_count=args.threads,
+        csv_filepath=args.csv_path,
+    )
+
+
+def run_dir_profiler(args):
+    logger.info("Testing files in the following directories:\n\t%s", "\n\t".join(args.dirs))
+    # Validate that provided directories exist
+    for dirpath in args.dirs:
+        if not os.path.isdir(dirpath):
+            raise Exception("Directory does not exist: %s" % dirpath)
+
+    filepaths = []
+    for dirpath in args.dirs:
+        filepaths.extend(get_files(dirpath, recurse=True))
+
+    profile(
+        filepaths,
+        stat=args.stat,
+        read=args.read,
+        hash_md5=args.md5,
+        hash_xx=args.xxhash if XXHASH else None,
+        warmup=args.warmup,
+        read_size=args.read_size,
+        thread_count=args.threads,
+        csv_filepath=args.csv_path,
+        skip_failures=args.skip_failures,
+    )
+
+
+def profile(filepaths, stat=True, read=True, hash_md5=True, hash_xx=True, warmup=True,
+            read_size=DEFAULT_READ_SIZE, thread_count=1, csv_filepath=None, skip_failures=False):
+
+    # Capture/record the arguments that were provided to the function. We'll log/write them out as
+    # part of the report.
+    args = inspect.getargvalues(inspect.currentframe())[3]
+    args["filepaths"] = len(args["filepaths"])
```

Comment:
Can you leave a comment explaining what this is for?

Reply:
yep, good call.